pkg/plugins/plugins_darwin.go

```diff
@@ -16,6 +16,20 @@ import (
 	"github.com/newrelic/infrastructure-agent/pkg/sysinfo/hostid"
 )
 
+func heartbeatOnlyMode(a *agent.Agent) bool {
+	config := a.GetContext().Config()
+	if !config.IsSecureForwardOnly && !config.IntegrationsOnly {
+		return false
+	}
```

Comment:
I would add some tests to assert that heartbeat is sent only in the expected modes.

Reply:
Added on last fix